test/integration/sds_static_integration_test.cc

```diff
@@ -0,0 +1,192 @@
+      validation_context->mutable_trusted_ca()->set_filename(
+          TestEnvironment::runfilesPath("test/config/integration/certs/cacert.pem"));
+      validation_context->add_verify_certificate_hash(
+          "E0:F3:C8:CE:5E:2E:A3:05:F0:70:1F:F5:12:E3:6E:2E:"
+          "97:92:82:84:A2:28:BC:F7:73:32:D3:39:30:A1:B6:FD");
+
+      common_tls_context->add_tls_certificate_sds_secret_configs()->set_name("server_cert");
+
+      auto* secret = bootstrap.mutable_static_resources()->add_secrets();
+      secret->set_name("server_cert");
+      auto* tls_certificate = secret->mutable_tls_certificate();
+      tls_certificate->mutable_certificate_chain()->set_filename(
+          TestEnvironment::runfilesPath("/test/config/integration/certs/servercert.pem"));
+      tls_certificate->mutable_private_key()->set_filename(
+          TestEnvironment::runfilesPath("/test/config/integration/certs/serverkey.pem"));
+    });
+
+    HttpIntegrationTest::initialize();
+
+    registerTestServerPorts({"http"});
+
+    client_ssl_ctx_ =
+        createClientSslTransportSocketFactory(false, false, context_manager_, secret_manager_);
+  }
+
+  void TearDown() override {
+    client_ssl_ctx_.reset();
+    cleanupUpstreamAndDownstream();
+    fake_upstream_connection_.reset();
+    codec_client_.reset();
+  }
+
+  Network::ClientConnectionPtr makeSslClientConnection() {
+    Network::Address::InstanceConstSharedPtr address = getSslAddress(version_, lookupPort("http"));
+    return dispatcher_->createClientConnection(address, Network::Address::InstanceConstSharedPtr(),
+                                               client_ssl_ctx_->createTransportSocket(), nullptr);
+  }
+
+private:
+  Runtime::MockLoader runtime_;
+  Ssl::ContextManagerImpl context_manager_{runtime_};
+  Secret::MockSecretManager secret_manager_;
+
+  Network::TransportSocketFactoryPtr client_ssl_ctx_;
+};
+
+INSTANTIATE_TEST_CASE_P(IpVersions, SdsStaticDownstreamIntegrationTest,
+                        testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                        TestUtility::ipTestParamsToString);
+
+TEST_P(SdsStaticDownstreamIntegrationTest, RouterRequestAndResponseWithGiantBodyBuffer) {
+  ConnectionCreationFunction creator = [&]() -> Network::ClientConnectionPtr {
+    return makeSslClientConnection();
+  };
+  testRouterRequestAndResponseWithBody(16 * 1024 * 1024, 16 * 1024 * 1024, false, &creator);
+}
+
+class SdsStaticUpstreamIntegrationTest
+    : public HttpIntegrationTest,
+      public testing::TestWithParam<Network::Address::IpVersion> {
+public:
+  SdsStaticUpstreamIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, GetParam()) {}
+
+  void initialize() override {
+    config_helper_.addConfigModifier([](envoy::config::bootstrap::v2::Bootstrap& bootstrap) {
+      auto* static_resources = bootstrap.mutable_static_resources();
+      auto* cluster = static_resources->mutable_clusters(0);
```

Comment:
nit: you don't need this line, can be combined with the line below without variable.

Reply:
Done